src/main.tsx

```diff
@@ -51,10 +61,34 @@ const Component = () => {
 };
 
 const Main = () => {
-  const problemDetails = useAppSelector(errorSelectors.all);
   const notification = useAppSelector(selectNotification);
   const dispatch = useAppDispatch();
+  const auth = useSelector((state: RootState) => state.auth);
```

Comment:
I don't think this is the correct way.

Both Refine and SaaS use AuthProvider as a ContextProvider and hooks, to avoid this.

Please revisit those library how they hook the Provider in the React components.

Reply:
sorry, wrong line. this is fine. is the useEffect() below that I was referring to.